layers/core_checks/cc_copy_blit_resolve.cpp

```diff
@@ -2452,16 +2454,19 @@ bool CoreChecks::ValidateImageBufferCopyMemoryOverlap(const CMD_BUFFER_STATE &cb
             DispatchGetImageSubresourceLayout(device, image_state.image(), &isr, &srl);
             if (image_state.createInfo.arrayLayers == 1) srl.arrayPitch = 0;
             if (image_state.createInfo.imageType != VK_IMAGE_TYPE_3D) srl.depthPitch = 0;
-            image_offset = (region.imageSubresource.baseArrayLayer * srl.arrayPitch) + (region.imageOffset.x * element_size) +
-                           (region.imageOffset.y * srl.rowPitch) + (region.imageOffset.z * srl.depthPitch) + srl.offset;
+            image_offset = (region.imageSubresource.baseArrayLayer * srl.arrayPitch) +
+                           static_cast<VkDeviceSize>((region.imageOffset.x * texel_size)) + (region.imageOffset.y * srl.rowPitch) +
+                           (region.imageOffset.z * srl.depthPitch) + srl.offset;
         } else {
-            image_offset = region.imageOffset.x * region.imageOffset.y * region.imageOffset.z * element_size;
+            image_offset =
+                static_cast<VkDeviceSize>(region.imageOffset.x * region.imageOffset.y * region.imageOffset.z * texel_size);
         }
-        uint64_t copy_size;
+        VkDeviceSize copy_size;
         if (region.bufferRowLength != 0 && region.bufferImageHeight != 0) {
-            copy_size = ((region.bufferRowLength * region.bufferImageHeight) * element_size);
+            copy_size = static_cast<VkDeviceSize>(((region.bufferRowLength * region.bufferImageHeight) * texel_size));
```

Comment:
```suggestion
            copy_size = static_cast<VkDeviceSize>(region.bufferRowLength) * region.bufferImageHeight * texel_size;
```
Assuming this will implicitly cast `region.bufferImageHeight` and `texel_size` to `VkDeviceSize` before doing the multiplication

Reply:
The cast is to handle texel_size being a double.  I believe all of the arithmetic will be done in floating point and at the end, cast to VkDeviceSize (uint64_t), which is what I think I want.  Taking the suggestion would result in a conversion, possible loss of data error